src/main/java/io/axoniq/axonserver/connector/query/impl/QueryChannelImpl.java

```diff
@@ -268,7 +268,12 @@ public Registration registerQueryHandler(QueryHandler handler, QueryDefinition..
                         } else {
                             future.complete(null);
                         }
-                        s.onNext(subscribeMessage);
+                        try {
+                            s.onNext(subscribeMessage);
+                        } catch (Exception e) {
+                            instructions.remove(subscribeMessage.getInstructionId());
```

Comment:
Ideally, to avoid any race conditions, it would be best to use the `remove(Object, Object)` method, to ensure that the future we remove from the instruction map is the one we just added. It's unlikely that the same instruction ID is used twice, but not impossible.